### ISSUE TITLE
"None" - gradient on "CUDA" device with no errors, but it work on "CPU" device.

### ISSUE THREAD

**Issue by PieceOfGood** [pytorch/pytorch#22860]
## To Reproduce
This python code:

    import torch
    
    print( torch.__version__ )
    print( torch.cuda.is_available() )
    print( torch.version.cuda )
Return to console:

    1.1.0
    True
    10.0
And this example code:

    import torch
    
    x = torch.tensor(
        [[1.,  2.,  3.,  4.],
         [5.,  6.,  7.,  8.],
         [9., 10., 11., 12.]], requires_grad=True)
    
    #######
    device = torch.device('cuda:0' 
                          if torch.cuda.is_available() 
                          else 'cpu')
    x = x.to(device)
    #######
    
    function = 10 * (x ** 2).sum()
    
    function.backward()
    
    print(x.grad, '<- gradient')
Return to console only this:

    None <- gradient
And no error messages. But, if I change

    device = torch.device('cuda'
                          if torch.cuda.is_available()
                          else 'cpu')
To

    device = torch.device('cpu')
Return to console:

    tensor([[ 20.,  40.,  60.,  80.],
            [100., 120., 140., 160.],
            [180., 200., 220., 240.]]) <- gradient

## Environment

 - PyTorch Version (e.g., 1.0): 1.1.0
 - OS (e.g., Linux): Windows 10
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7.4
 - CUDA/cuDNN version: 10
 - GPU models and configuration: GeForce 1080 TI
 - Any other relevant information:

**Comment by SsnL** [comment]
`x = x.to(device)`

Means that if `device` is `x`'s current device, i.e., `CPU`, this does no-op, and keeps `x` intact. Otherwise, it does a copy and thus makes `x` an intermediate result. 

Intermediate results do not keep gradients, by default.
